lib/resourcedelete/apiext.go

```diff
@@ -0,0 +1,41 @@
+package resourcedelete
+
+import (
+	"context"
+	"fmt"
+
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextclientv1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeleteCustomResourceDefinitionv1 checks the given resource for a valid delete annotation. If found
+// and in UpdatingMode it will issue a delete request or provide status of a previousily issued delete request.
+// If not in UpdatingMode it simply returns an indication that the delete annotation was found. An error is
+// returned if an invalid annotation is found or an error occurs during delete processing.
+func DeleteCustomResourceDefinitionv1(ctx context.Context, client apiextclientv1.CustomResourceDefinitionsGetter,
+	required *apiextv1.CustomResourceDefinition, updateMode bool) (bool, error) {
+
+	if delAnnoFound, err := ValidDeleteAnnotation(required.Annotations); !delAnnoFound || err != nil {
+		return delAnnoFound, err
+	} else if !updateMode {
+		return true, nil
+	}
```

Comment:
In both the conditions, you are returning from the function. It is not clear not clear from code which situation the code will not meet these conditions and continue executing the rest of the function.

Reply:
In the first case, we either didn't find the delete annotation or we got an error. The only error we can get is that the delete annotation is found but doe not have value `true`. It must have value `true`. Otherwise, if a valid delete annotation is found we check that the cluster is actually in `UpdateMode` since this is the only time we should be performing the delete (see `lib/resourcebuilder/resourcebuilder.go`.

This is explained in the method comment.

Reply:
Makes sense. Thanks